src/segment-loader.js

```diff
@@ -1104,4 +1109,42 @@ export default class SegmentLoader extends videojs.EventTarget {
     this.throughput.rate +=
       (segmentProcessingThroughput - rate) / (++this.throughput.count);
   }
+
+  /**
+   * Adds a cue to the segment-metadata track with some metadata information about the
+   * segment
+   *
+   * @private
+   * @param {Object} segmentInfo
+   *        the object returned by loadSegment
+   * @method addSegmentMetadataCue_
+   */
+  addSegmentMetadataCue_(segmentInfo) {
+    if (!this.segmentMetadataTrack_ || !segmentInfo) {
+      return;
+    }
+
+    const playlist = segmentInfo.playlist;
+    const segment = segmentInfo.segment;
+    const start = segment.start;
+    const end = segment.end;
+
+    removeCuesFromTrack(start, end, this.segmentMetadataTrack_);
+
+    const value = {
+      uri: segmentInfo.uri,
+      timeline: segmentInfo.timeline,
+      playlist: playlist.uri,
+      start,
+      end
+    };
+    const Cue = window.WebKitDataCue || window.VTTCue;
+    const data = JSON.stringify(value);
```

Comment:
do you mean if say you wanted the uri, accessing it should look like `cue.uri` instead of `cue.value.uri`?